spec/ysh-methods.test.sh

```diff
@@ -552,6 +552,19 @@ pp test_ (book)
 (Dict)   {"title":"The Histories"}
 ## END
 
+#### Dict -> clear()
+var book = {title: "The Histories", author: "Herodotus"}
+call book->clear()
+pp test_ (book)
+# confirm method is idempotent
+call book->clear()
+pp test_ (book)
```

Comment:
Let's start using `assert [expected === actual]` where possible, as in the last PR

I know the tests are not consistent about this!  But it does make it easier to read